ArgusCore/src/main/java/com/salesforce/dva/argus/entity/Policy.java

```diff
@@ -0,0 +1,276 @@
+package com.salesforce.dva.argus.entity;
+
+import static com.salesforce.dva.argus.system.SystemAssert.requireArgument;
+
+import java.util.List;
+
+import javax.persistence.Basic;
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.EntityManager;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.NoResultException;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.persistence.TypedQuery;
+import javax.persistence.UniqueConstraint;
+
+import com.salesforce.dva.argus.entity.Trigger.TriggerType;
+import com.salesforce.dva.argus.util.Cron;
+
+/**
+ * The entity encapsulates information about the policy.
+ *
+ * <p>Fields that determine uniqueness are:</p>
+ *
+ * <ul>
+ *   <li>NAME</li>
+ *   <li>SERVICE</li>
+ * </ul>
+ *
+ * <p>Fields that cannot be null are:</p>
+ *
+ * <ul>
+ *   <li>SERVICE</li>
+ *   <li>NAME</li>
+ *   <li>OWNER</li>
+ *   <li>USER</li>
+ *   <li>SUB_SYSTEM</li>
+ *   <li>METRIC_NAME</li>
+ *   <li>TRIGGER_TYPE</li>
+ *   <li>AGGREGATOR</li>
+ *   <li>THRESHOLD</li>
+ *   <li>TIME_UNIT</li>
+ *   <li>DEFAULT_VALUE</li>
+ *   <li>CRON_ENTRY</li>
+ * </ul>
+ *
+ * @author  Ruofan Zhang (rzhang@salesforce.com)
+ */
+@SuppressWarnings("serial")
+@Entity
+@Table(name = "POLICY", uniqueConstraints = @UniqueConstraint(columnNames = { "name","service" }))
+@NamedQueries(
+	    {
+	        @NamedQuery(
+	            name = "Policy.findByName", query = "SELECT r FROM Policy r WHERE r.name = :name AND r.service = :service"
+	        )
+	    }
+	)
+public class Policy extends JPAEntity {
+	//~ Instance fields ******************************************************************************************************************************
+
+	@Basic(optional = false)
+	@Column(nullable = false)
+    private String service;
+	
+	@Basic(optional = false)
+	@Column(nullable = false)
+    private String name;
+	
+	@Basic(optional = false)
+	@Column(nullable = false)
+    @ElementCollection
+    private List<String> listOfOwner;
+	
+	@Basic(optional = false)
+	@Column(nullable = false)
+    @ElementCollection
+    private List<String> listOfUser;
+	
+	@Basic(optional = true)
+	@Column(name = "sub_system", nullable = false)
+    private String subSystem;
```

Comment:
Here you have it as optional which is correct, but you also have it marked as not nullable which is incorrect.  Subsystem is an optional field.